plugins/circleci/src/circleci-config.ts

```diff
@@ -1,29 +1,47 @@
 import { promises as fs } from 'fs'
-import * as yaml from 'js-yaml'
-import isEqual from 'lodash.isequal'
+import merge from 'lodash/merge'
 import path from 'path'
+import YAML from 'yaml'
 import { ToolKitError } from '@dotcom-tool-kit/error'
 import { styles } from '@dotcom-tool-kit/logger'
 import { getOptions } from '@dotcom-tool-kit/options'
 import { Hook } from '@dotcom-tool-kit/types'
-import { Workflow, JobConfig, CircleConfig, automatedComment } from '@dotcom-tool-kit/types/lib/circleci'
+import { semVerRegex } from '@dotcom-tool-kit/types/lib/npm'
+import { JobConfig, automatedComment } from '@dotcom-tool-kit/types/lib/circleci'
 
 const majorOrbVersion = '2'
 
-export default abstract class CircleCiConfigHook extends Hook {
+interface CircleCIState {
+  jobs: Record<string, JobConfig>
+  nightlyJobs: Record<string, JobConfig>
+  runOnVersionTags: boolean
+}
+
+const tagFilter = { filters: { tags: { only: `${semVerRegex}` } } }
+
+function isAutomatedConfig(config: YAML.Document): boolean {
+  const contents = config.contents as YAML.YAMLMap | undefined
+  const topKey = contents?.items[0].key as YAML.Scalar | undefined
+  const topComment = topKey?.commentBefore?.trim()
+  return topComment === automatedComment.slice(1).trim()
+}
```

Comment:
This is much less readable than the previous check for automated config; `rawConfig.startsWith(automatedComment)`, why is this better?

I know you've added `YAML` to simplify the code elsewhere but you could still convert this check to be on a string 
 for simplicity: `config?.toString().startsWith(automatedComment)`

Reply:
Yeah, I think you're right that it would be more readable to just run a `toString()` conversion here. It's a reasonable compromise from this convoluted approach that was trying to improve upon the previous convoluted approach!